### ISSUE TITLE
Diagnostics false negatives

### ISSUE THREAD

**Issue by Strum355** [fwcd/KotlinLanguageServer#47]
I have the following line (from this repo) `return Hover(listOf(hover), range)` for which KLS reports:
```
[INFO]      debounce0   main    Linting .../hover/hovers.kt
[INFO]      async0      main    Finished in 11 ms
[INFO]      debounce0   main    No diagnostics in hovers.kt
[INFO]      async0      main    Find symbols in TextDocumentIdentifier [
                                uri = "file:///home/noah/Kotlin/KotlinLanguageServer/src/main/kotlin/org/javacs/kt/hover/hovers.kt"
                                ]    
```

while gradle complains:
`e: /home/noah/Kotlin/KotlinLanguageServer/src/main/kotlin/org/javacs/kt/hover/hovers.kt: (51, 74): Expecting ')'`

**Comment by fwcd** [comment]
@Strum355  I do not see any issues with this line:

https://github.com/fwcd/KotlinLanguageServer/blob/cb946ecc73319b319f53e037588187f5b90d1054/src/main/kotlin/org/javacs/kt/hover/hovers.kt#L51

The Gradle build did pass too: https://travis-ci.org/fwcd/KotlinLanguageServer/builds/409471711

Maybe you have edited the file unintentionally?

**Comment by Strum355** [comment]
yea thats the point, i had edited myself but KotlinLangServer didnt give a diagnostic for it, the output as shown in the codeblock also says that theres no diagnostics

**Comment by fwcd** [comment]
@Strum355 The linter ist not 100% accurate currently, because the mechanism used to incrementally compile an expression [is somewhat hacky](https://github.com/fwcd/KotlinLanguageServer/blob/master/README.md#this-repository-needs-your-help).

Maybe some kind of race condition occurred when the compiler tries to evaluate your currently selected expression as you move your cursor too quickly for the linter to keep up.

I will leave this open for now.

**Comment by fwcd** [comment]
@Strum355 Yes, they all run on a single async thread (except for the linter), because the Kotlin compiler internals seems to have trouble with concurrency (see #42).
